cptv-download.py

```diff
@@ -69,21 +69,29 @@ def process(self, url):
         print("Dates are {0} - {1}".format(self.start_date, self.end_date))
         print("Required tags are {0}".format(self.only_tags))
         print("Ignore tags are {0}".format(self.ignore_tags))
+        pool = Pool(self.workers, self._downloader, api, Path(self.out_folder))
+        offset = 0
+        limit = self.limit
+        while self.limit is None or offset < self.limit:
+            rows = api.query(
+                limit=limit,
+                startDate=self.start_date,
+                endDate=self.end_date,
+                tagmode=self.tag_mode,
+                tags=self.only_tags,
+                offset=offset,
+            )
+            if len(rows) == 0:
+                break
+            offset += len(rows)
+            if limit:
```

Comment:
You are increasing the offset AND reducing the limit.   Won't this result in getting half as many recordings as requested?

Reply:
I don't think so, the limit at this point is the number of remaining recordings. 
e.g. if you run the program with limit 200, and lets say it only gets 100 rows at a time.
The first download requeust will be offset 0 limit 200 - returns 100
The second download request will be offset 100 limit 100( as we only want 200 total) -returns the remaining 100

Reply:
But the second request won't run because offset < self.limit will fail.   This condition should probably be self.limit > 0 and then it will work.

Reply:
ah i see, its confusing but there is 2 variables,
`limit` and `self.limit`, so `limit` decreases it is actually the reamining count i'll rename that, while 'self.limit` remains at 200 (or whatever is input by the user)